### PR TITLE
[Sqlite Adapter] ADDED migrationEvents

### PR DESCRIPTION
**Pull request by avinashlng1080**
Allowed callbacks to be triggered for when the migration process starts, fails and succeeds.

### CODE REVIEW

File: src/adapters/sqlite/index.js [hunk 1]
```diff
@@ -157,8 +162,10 @@ export default class SQLiteAdapter implements DatabaseAdapter, SQLDatabaseAdapte
           ),
         )
         logger.log('[WatermelonDB][SQLite] Migration successful')
+        this.migrationEvents?.onSuccess?.()
       } catch (error) {
         logger.error('[WatermelonDB][SQLite] Migration failed', error)
+        this.migrationEvents?.onFailure?.(error)
```
Comment (radex) [inline src/adapters/sqlite/index.js:168]: sorry, but no `?.(` syntax please - it breaks some of our tooling

File: src/adapters/sqlite/index.js [hunk 2]
```diff
@@ -48,6 +49,8 @@ export default class SQLiteAdapter implements DatabaseAdapter, SQLDatabaseAdapte
 
   migrations: ?SchemaMigrations
 
+  migrationEvents: ?MigrationEvents
```
Comment (radex) [inline src/adapters/sqlite/index.js:52]:
```suggestion
  _migrationEvents: ?MigrationEvents
```

File: src/adapters/sqlite/type.js [hunk 3]
```diff
@@ -13,12 +13,19 @@ export type SQL = string
 export type SQLiteArg = string | boolean | number | null
 export type SQLiteQuery = [SQL, SQLiteArg[]]
 
+export type MigrationEvents = {
+  onSuccess?: () => void,
+  onStarted?: () => void,
+  onFailure?: (error: string) => void,
```
Comment (radex) [inline src/adapters/sqlite/type.js:19]: I think for simplification (no `?.()`), those can be required. So, if you're passing migrationEvents at all, you must specify all three.

**Review by radex — CHANGES_REQUESTED** [review]
a few changes please ;)

File: src/adapters/sqlite/index.js (hunk 1 above)
Comment (avinashlng1080) [reply src/adapters/sqlite/index.js]: corrected.

File: src/adapters/sqlite/index.js (hunk 2 above)
Comment (avinashlng1080) [reply src/adapters/sqlite/index.js]: corrected.

File: src/adapters/sqlite/type.js (hunk 3 above)
Comment (avinashlng1080) [reply src/adapters/sqlite/type.js]: corrected.

**Comment by radex** [comment]
Take a look at the CI error https://travis-ci.com/github/Nozbe/WatermelonDB/jobs/452491959

you can do `yarn ci:check` locally to run those same checks and make sure there's no errors

**Pull request merged**